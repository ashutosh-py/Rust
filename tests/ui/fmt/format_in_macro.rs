// aux-build:format_in_macro.rs
extern crate format_in_macro;
#[derive(format_in_macro::derive)]
pub struct Foo;
//~^^ERROR format argument must be a string literal

format_in_macro::function!();
//~^ERROR format argument must be a string literal

#[format_in_macro::attribute]
const UNIT:() = ();
//~^^ERROR format argument must be a string literal

macro_rules! indirect {
    () => {
        format_in_macro::function!();
        //~^ERROR format argument must be a string literal
    }
}
fn new_scope() {
    indirect!();
}

fn main() {}
