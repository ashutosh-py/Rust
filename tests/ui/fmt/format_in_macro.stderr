error: format argument must be a string literal
  --> $DIR/format_in_macro.rs:3:10
   |
LL | #[derive(format_in_macro::derive)]
   |          ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `format_in_macro::derive` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you might be missing a string literal to format with
   |
LL | #[derive("{}", format_in_macro::derive)]
   |          +++++

error: format argument must be a string literal
  --> $DIR/format_in_macro.rs:7:1
   |
LL | format_in_macro::function!();
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `format_in_macro::function` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you might be missing a string literal to format with
   |
LL | "{}", format_in_macro::function!();
   | +++++

error: format argument must be a string literal
  --> $DIR/format_in_macro.rs:10:1
   |
LL | #[format_in_macro::attribute]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `format_in_macro::attribute` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you might be missing a string literal to format with
   |
LL | "{}", #[format_in_macro::attribute]
   | +++++

error: format argument must be a string literal
  --> $DIR/format_in_macro.rs:16:9
   |
LL |         format_in_macro::function!();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
LL |     indirect!();
   |     ----------- in this macro invocation
   |
   = note: this error originates in the macro `format_in_macro::function` which comes from the expansion of the macro `indirect` (in Nightly builds, run with -Z macro-backtrace for more info)
help: you might be missing a string literal to format with
   |
LL |         "{}", format_in_macro::function!();
   |         +++++

error: aborting due to 4 previous errors

